scripts/live-connector-e2e/report.py

```
@@ -56,36 +96,173 @@ def load_results(results_dir: Path, *, roots: list[Path] | None = None) -> list[
+
+
+def _matches_candidate_failure(row: dict, item: Classification) -> bool:
+    return (
+        row.get("connector") == item.connector
+        and row.get("status") == "fail"
+        and str(row.get("event", "")).startswith("candidate-")
+        and str(row.get("version") or "unknown") == item.candidate_version
+    )
+
+
+def load_classifications(results_dir: Path) -> list[Classification]:
```

Ruff (0.16.1)

[warning] Too many branches (14 > 12)

(PLR0912)